pkg/cmd/cli/cmd/logs.go

```diff
@@ -189,5 +218,32 @@ func (o OpenShiftLogsOptions) RunLog() error {
 		// Use our own options object.
 		o.KubeLogOptions.Options = o.Options
 	}
-	return o.KubeLogOptions.RunLogs()
+	if !o.JenkinsLogs {
+		return o.KubeLogOptions.RunLogs()
+	}
+
+	bc, ok := o.KubeLogOptions.Object.(*buildapi.BuildConfig)
+	var urlString string
+	if !ok {
+		build, ok := o.KubeLogOptions.Object.(*buildapi.Build)
+		if ok {
+			urlString, _ = build.Annotations[buildapi.BuildJenkinsBlueOceanLogURLAnnotation]
+		}
+	} else {
+		buildName := buildutil.BuildNameForConfigVersion(bc.ObjectMeta.Name, int(bc.Status.LastVersion))
+		build, _ := o.Client.Builds(o.Namespace).Get(buildName, metav1.GetOptions{})
+		if build != nil {
+			urlString, _ = build.Annotations[buildapi.BuildJenkinsBlueOceanLogURLAnnotation]
+		}
+	}
+	if len(urlString) > 0 {
+		o.KubeLogOptions.Out.Write([]byte("Because this is a Pipeline Strategy build, the logs are accessed via the Jenkins server:"))
+		o.KubeLogOptions.Out.Write([]byte("\n"))
+		o.KubeLogOptions.Out.Write([]byte(urlString))
+		o.KubeLogOptions.Out.Write([]byte("\n"))
+	} else {
+		return errors.New("Could not find the log URL for this Jenkins Pipeline Strategy build")
```

Comment:
On the second part of the comment, if we did everything in `RunLogs()` we would have to duplicate the arg parsing that occurs out of `Complete()` to determine whether we are even dealing with builds/buildconfigs.  Piggybacking on top of that looked cleaner when I played around with this.

I'll break up the errors into more fine grained instances.